web/modules/custom/hoeringsportal_data/src/Controller/StatisticsController.php

```diff
@@ -0,0 +1,310 @@
+    private readonly EventHelper $pretixEventHelper,
+    #[Autowire(service: 'hoeringsportal_deskpro.helper')]
+    private readonly HearingHelper $hearingHelper,
+    private readonly UrlGeneratorInterface $urlGenerator,
+    #[Autowire(service: 'serializer')]
+    private readonly SerializerInterface $serializer,
+    #[Autowire(service: 'file.mime_type.guesser')]
+    private readonly MimeTypeGuesserInterface $mimeTypeGuesser,
+  ) {
+  }
+
+  /**
+   * Index.
+   */
+  public function index(Request $request) {
+    $data = $this->getData($request);
+
+    if (NULL !== $data) {
+      [$report, $data] = $data;
+      $format = $request->query->get('export');
+      if (NULL !== $format) {
+        return $this->export($report, $data, $format);
+      }
+    }
+
+    $form = $this->formBuilder()->getForm(StatisticsForm::class, ['reports' => $this->getReports()]);
+    unset($form['build_id']);
+
+    return [
+      'form' => $form,
+      'result' => [
+        '#theme' => 'hoeringsportal_data_statistics_export',
+        '#data' => $data,
+        '#options' => [
+          'url_field' => (string) $this->t('Url'),
+          'title_field' => (string) $this->t('Title'),
+        ],
+      ],
+    ];
+  }
+
+  /**
+   * Export data.
+   */
+  private function export(array $report, array $data, string $format = 'csv'): Response {
+    $filename = 'export.' . $format;
+    if (isset($report['export_filename_template'])) {
+      $tokens = [
+        'date' => (new \DateTimeImmutable())->format('Y-m-d\TH-i-s'),
+        'title' => $report['title'],
+        'format' => $format,
+      ];
+      // Replace placeholder on the form `{key}` in file name template.
+      $filename = preg_replace_callback(
+        '/\{([^}]+)\}/',
+        static fn (array $matches): string => $tokens[$matches[1]] ?? $matches[0],
+        $report['export_filename_template'],
+      );
+    }
+    $response = new Response($this->serializer->serialize($data, $format));
+
+    $contentType = $this->mimeTypeGuesser->guessMimeType($filename);
+    if (NULL !== $contentType) {
+      $response->headers->set(
+        'content-type',
+        'text/csv'
+      );
+    }
+
+    $response->headers->set(
+      'content-disposition',
+      $response->headers->makeDisposition(
+        ResponseHeaderBag::DISPOSITION_ATTACHMENT,
+        $filename
+      )
+    );
+
+    return $response;
+  }
+
+  /**
+   * Get data.
+   */
+  private function getData(Request $request): ?array {
+    $values = $request->query->all();
+    $requestedReport = $request->query->get(StatisticsForm::REPORT);
+    $reports = $this->getReports();
+    $report = $reports[$requestedReport] ?? NULL;
+
+    if ($requestedReport && !isset($report['callback'])) {
+      throw new NotFoundHttpException();
+    }
+    $run = isset($values['show']) || isset($values['export']);
+    if ($run) {
+      $startDate = new \DateTimeImmutable('first day of january');
```

Comment:
```suggestion
      $startDate = new \DateTimeImmutable('first day of January');
```

Reply:
The grammar (https://www.php.net/manual/en/datetime.formats.php#datetime.formats.date) tells me to use `january`, but the examples (https://www.php.net/manual/en/datetime.formats.php#datetime.formats.relative) use `January`. I'll use January …